xds/internal/balancer/lrs/lrs_test.go

```diff
@@ -13,505 +13,159 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ *
  */
 
 package lrs
 
 import (
-	"context"
 	"fmt"
-	"io"
-	"net"
-	"sort"
-	"sync"
 	"testing"
 	"time"
 
-	corepb "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	endpointpb "github.com/envoyproxy/go-control-plane/envoy/api/v2/endpoint"
-	lrsgrpc "github.com/envoyproxy/go-control-plane/envoy/service/load_stats/v2"
-	lrspb "github.com/envoyproxy/go-control-plane/envoy/service/load_stats/v2"
-	"github.com/golang/protobuf/proto"
-	durationpb "github.com/golang/protobuf/ptypes/duration"
-	structpb "github.com/golang/protobuf/ptypes/struct"
 	"github.com/google/go-cmp/cmp"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	"google.golang.org/grpc/xds/internal"
+	"google.golang.org/grpc/attributes"
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/balancer/roundrobin"
+	"google.golang.org/grpc/connectivity"
+	internalserviceconfig "google.golang.org/grpc/internal/serviceconfig"
+	"google.golang.org/grpc/resolver"
+	xdsinternal "google.golang.org/grpc/xds/internal"
 	"google.golang.org/grpc/xds/internal/testutils"
 )
 
-const (
-	testService             = "grpc.service.test"
-	testHostname            = "grpc.server.name"
-	nodeMetadataHostnameKey = "PROXYLESS_CLIENT_HOSTNAME"
-)
-
 var (
-	dropCategories = []string{"drop_for_real", "drop_for_fun"}
-	localities     = []internal.LocalityID{{Region: "a"}, {Region: "b"}}
-	errTest        = fmt.Errorf("test error")
+	testBackendAddrs = []resolver.Address{
+		{Addr: "1.1.1.1:1"},
+	}
+	testLocality = &xdsinternal.LocalityID{
+		Region:  "test-region",
+		Zone:    "test-zone",
+		SubZone: "test-sub-zone",
+	}
 )
 
-type rpcCountDataForTest struct {
-	succeeded   uint64
-	errored     uint64
-	inProgress  uint64
-	serverLoads map[string]float64
+// This is a subset of testutils.fakeclient. Cannot use testutils.fakeclient
+// because testutils imports package lrs.
```

Comment:
The `fakeClient` depends on `package lrs` because of the `lrs.Store` type. I think we might have to pull that out of `lrs package` anyways because we need to add v3 support for LRS as well. If we had `package lrs/store`, we wouldn't have this issue here. Maybe add a TODO here for the same?

Reply:
Done